### ISSUE TITLE
./mwaa-local-env build-image certificate problem

### ISSUE THREAD

**Issue by jcdennis** [aws/aws-mwaa-local-runner#186]
In the step where I run "./mwaa-local-env build-image" I am receiving the following error:

https://cdn.amazonlinux.com/2/core/2.0/x86_64/0cac95da63306270fbedb235008cfaf4f04477723dc1966e2abadb274a0edd44/repodata/repomd.xml?instance_id=timeout&region=unknown: [Errno 14] curl#60 - "SSL certificate problem: unable to get local issuer certificate"
------
executor failed running [/bin/sh -c chmod u+x /systemlibs.sh && /systemlibs.sh]: exit code: 1

Any advice on how to solve?

**Comment by anupash147** [comment]
am getting it too..

**Comment by DASHSA2** [comment]
https://cdn.amazonlinux.com/2/core/2.0/x86_64/0cac95da63306270fbedb235008cfaf4f04477723dc1966e2abadb274a0edd44/repodata/repomd.xml?instance_id=URLError&region=unknown: [Errno 14] curl#60 - "SSL certificate problem: self signed certificate in certificate chain"
 
 It appears the instance_id and region of the repo server cannot be found along with a self-signed certificate. Please comment how to move forward as this is occurring due to execution of this command
 
 RUN chmod u+x /systemlibs.sh && /systemlibs.sh in Docker file.

When the packages are being downloaded by executing yum commands in systemlibs.sh file this issue is occurring.

**Comment by ghost** [comment]
I believe this is an issue with the `curl` command on line 51 of the docker/script/bootstrap.sh.  It is trying to curl (retrieve) a file from a server.  You should be able to fix the issue by adding the destination URLs certificate to your SSL certificate repository on your computer.  

You can learn how to download an SSL certificate from "https://awscli.amazonaws.com" from these instructions [https://mattferderer.com/fix-git-self-signed-certificate-in-certificate-chain-on-windows](https://mattferderer.com/fix-git-self-signed-certificate-in-certificate-chain-on-windows).  

For apple users, follow the same instructions for downloading the certificate, but then you need to attach it to your keychain - I think these instructions will help [https://uwm.edu/iam/client-certificate-install-on-mac/](https://uwm.edu/iam/client-certificate-install-on-mac/).

Basically, you need to have the destination server's SSL certificate in your chain of trusted certificates.  Once you get that installed you should be able to build the image.  

Good luck and please update on whether it worked!  (not a contributor, but we faced the same issue this week).

**Comment by DASHSA2** [comment]
After downloading the certificates, How are you installing them where are you installing them in case of windows local machine ?

**Comment by DASHSA2** [comment]
I tried the solution by installing certificates, but it still doesn't work. Primarily the problem persists as there is a dependency "yum install libaio" when I do it in a corporate set up it gives me a self-signed certificate chain error but when I do it on my personal machine it is delayed but it works. Will whitelisting Amazon Linux 2 repository work?
